api/src/Repository/DepartmentRepository.php

```diff
@@ -116,7 +117,23 @@ public function update(/*.string.*/$id, /*.mixed.*/$data): void
 
     public function deleteById(/*.mixed.*/$id): void
     {
-        throw new Exception(__CLASS__.": Method '__FUNCTION__' not implemented");
+        $placeholderDeptId = Select::new($this->db)
```

Comment:
Seems like I may need to add a few things here:

- Delete from EMails on DepartmentID
- Delete from EmailListAccess on DepartmentID
- Update other department FallbackIDs to null on DepartmentID

May be able to clean up the Placeholder Dept ID lookup

Reply:
yeah, we will need to handle all those.  Just removing the emails and such should be fine. One thing the current code does not do cleanly is handle staff assigned to a department being deleted. Presently it is just operation procedure that there needs to be zero staff assigned to the department before it is deleted.  We could enforce this in code which may be good.

Reply:
I like the idea of enforcing that in code. I will make sure to update this to handle the other cases for now. I'll check on whether we can handle staff on a deleted department as well. 

Do the staff that are deleted from one department get added to other departments typically? Would we want to be able to keep track of years served?

Reply:
Generally yes, they are moved to other departments.  But again it is a soft policy thing. I am sure there would be a case at some point where a staff member will leave or such.

And yes tracking years served is important.